### PR TITLE
feat(pagination): harden PaginationCacheUtil typing and add unit tests

### LINKED ISSUE #99: [TASK] Harden PaginationCacheUtil typing and add unit tests for cache keys
## Overview

`PaginationCacheUtil` (`src/pagination/utils/PaginationCacheUtil.ts`) is used by list endpoints (e.g. store, product) to build deterministic Redis/cache keys from filter objects. It currently uses `Record<string, any>` and `any` in several signatures, which conflicts with project rules (avoid `any` in production code). There is also **no dedicated unit test** for `buildCacheKey` / `sanitizeFilters`, so regressions in key stability or nesting behavior are easy to miss.

This task is a focused exercise in **strict typing**, **deterministic serialization**, and **Jest unit tests** without changing HTTP contracts or cache semantics.

## Steps

1. Read `PaginationCacheUtil` and call sites (`store.service.ts`, `product.service.ts`) to confirm which public methods must remain backward-compatible.
2. Replace `any` with appropriate types (`Record<string, unknown>`, generics, or narrow interfaces) for `buildCacheKey`, `sanitizeFilters`, and related helpers. For `applyPagination` / `invalidateCache`, use the smallest correct types (e.g. `SelectQueryBuilder` from TypeORM where applicable, or a minimal interface if that avoids pulling heavy deps into tests).
3. Add `src/pagination/utils/pagination-cache-util.spec.ts` (or colocated name matching project convention) covering at least:
   - **Determinism:** two filter objects with the same keys/values but different property insertion order produce the **same** cache key string.
   - **Nesting:** nested objects are sanitized consistently; depth limit (`depth > 3`) is respected.
   - **Edge cases:** `null`/`undefined` values stripped; empty filters; invalid inputs throw or behave as documented today.
4. Run the full verification pipeline from `AGENTS.md` / `CLAUDE.md`.

## Acceptance criteria

- [ ] `PaginationCacheUtil` production paths do not use `any` without an explicit `eslint-disable-next-line` and a one-line justification (prefer zero disables).
- [ ] New unit tests pass and meaningfully assert cache key stability and sanitizer behavior (not only “does not throw”).
- [ ] `yarn build`, `yarn lint`, `yarn test`, and `yarn format:check` pass.
- [ ] No change to the **hash algorithm** or external cache key strings for identical inputs (behavior preserved for ASCII-only filter keys in use today).

## Technical context

- **Primary file:** `src/pagination/utils/PaginationCacheUtil.ts`
- **Consumers:** `src/store/store.service.ts`, `src/product/product.service.ts`
- **References:** `AGENTS.md` (no `any`), `docs/CONVENTIONS.md`

## Out of scope

- Stripe, subscriptions, or payment flows.
- Changing `CacheService` behavior, TTL, or invalidation strategy.
- End-to-end or integration tests against Redis (unit tests only).
- Replacing `createHash('sha256')` or JSON serialization strategy.

## Learning goals (for the assignee)

- Practice replacing `any` with `unknown` and safe narrowing.
- Understand why **stable key ordering** matters for cache hits.
- Write focused Jest tests for a static utility class.

### PR DESCRIPTION
**Pull request by cedenolozanoazalia-hub**
Replace any with unknown and Record<string, unknown>
Add PaginationMetrics interface with optional properties Type applyPagination with SelectQueryBuilder<T>
Type invalidateCache with CacheService
Add comprehensive unit tests for cache key determinism Add tests for null/undefined filtering and nested objects

## What & Why

The `PaginationCacheUtil` class uses `any` in several signatures, which conflicts with project rules. This PR replaces `any` with proper types and adds missing unit tests to prevent regressions in cache key stability.
Closes #99

## Changes Summary

- Replace `any` with `unknown` and `Record<string, unknown>` in `buildCacheKey`, `sanitizeFilters`
- Add `PaginationMetrics` interface with optional properties for metrics collector
- Type `applyPagination` with `SelectQueryBuilder<T>` instead of `any`
- Type `invalidateCache` with `CacheService` instead of `any`
- Add `pagination-cache-util.spec.ts` with 13 unit tests covering:
  - Deterministic cache keys (order-independent)
  - Null/undefined filtering
  - Nested object sanitization
  - Depth limit (max 3 levels)
  - Array handling

 ## Type of Change

- [ ] `feat` — New feature
- [ ] `fix` — Bug fix
- [x] `refactor` — Code change that neither fixes a bug nor adds a feature
- [x] `test` — Adding or updating tests
- [ ] `chore` — Maintenance, config, tooling

## Definition of Done

See [DEFINITION_OF_DONE.md](../DEFINITION_OF_DONE.md) for full checklist.

- [x] **PR title** follows Conventional Commits: `type(scope): description` (max 72 chars)
- [ ] **Link issue:** Body includes `Closes #N` or `Fixes #N` to auto-close on merge
- [x] Code passes `yarn quality` (build, lint, test, format:check)
- [x] No `any` in production code; explicit return types on service methods
- [x] Unit tests for new/changed business logic
- [ ] Swagger docs updated for API changes
- [ ] No unresolved TODO comments
- [ ] **ADR created** if architectural decision was made ([docs/adr/000-index.md](../docs/adr/000-index.md)) — checkbox below if applicable

## Architectural Decision

- [ ] No architectural decision in this PR
- [ ] ADR created and linked: [docs/adr/NNNN-short-title.md](../docs/adr/)

## AI Assistance Disclosure

- [ ] No AI assistance used
- [ ] AI assistance used — reviewed and understood

## Breaking Changes

None

## Testing Instructions for Reviewer

1. Run `yarn test -- src/pagination/utils/pagination-cache-util.spec.ts` (13 tests should pass)
2. Run `yarn build` (should compile without errors)
3. Run `yarn lint:check` (should pass)
4. Verify that existing list endpoints (store, product) still work with cache

### CODE REVIEW

File: src/pagination/utils/PaginationCacheUtil.ts [hunk 1]
```diff
@@ -22,32 +41,59 @@ export class PaginationCacheUtil {
...
+  ): Record<string, unknown> {
+    // Limitar profundidad máxima para evitar recursión excesiva
     if (depth > 3) return {};
 
-    return Object.keys(filters)
-      .sort()
-      .reduce((obj, key) => {
-        const value = filters[key];
-        if (value === undefined || value === null) return obj;
+    const sortedKeys = Object.keys(filters).sort();
```
Comment (RaulAltamirano) [inline src/pagination/utils/PaginationCacheUtil.ts]: Provide a compare function that depends on "String.localeCompare", to reliably sort elements alphabetically.
